digdag-standards/src/main/java/io/digdag/standards/command/EcsCommandExecutor.java

```diff
@@ -0,0 +1,685 @@
+    private static final String DEFAULT_COMMAND_TASK_TTL = ECS_COMMAND_EXECUTOR_SYSTEM_CONFIG_PREFIX + "default_command_task_ttl";
+
+    private final Config systemConfig;
+    private final EcsClientFactory ecsClientFactory;
+    private final DockerCommandExecutor docker;
+    private final StorageManager storageManager;
+    private final ProjectArchiveLoader projectArchiveLoader;
+    private final CommandLogger clog;
+    private final Optional<Duration> defaultCommandTaskTTL;
+
+    @Inject
+    public EcsCommandExecutor(
+            final Config systemConfig,
+            final EcsClientFactory ecsClientFactory,
+            final DockerCommandExecutor docker,
+            final StorageManager storageManager,
+            final ProjectArchiveLoader projectArchiveLoader,
+            final CommandLogger clog)
+    {
+        this.systemConfig = systemConfig;
+        this.ecsClientFactory = ecsClientFactory;
+        this.docker = docker;
+        this.storageManager = storageManager;
+        this.projectArchiveLoader = projectArchiveLoader;
+        this.clog = clog;
+        this.defaultCommandTaskTTL = systemConfig.getOptional(DEFAULT_COMMAND_TASK_TTL, DurationParam.class)
+                .transform(DurationParam::getDuration)
+                .or(Optional.absent());
+    }
+
+    @Override
+    public CommandStatus run(
+            final CommandContext commandContext,
+            final CommandRequest commandRequest)
+            throws IOException
+    {
+        final Config taskConfig = commandContext.getTaskRequest().getConfig();
+        final Optional<String> clusterName = Optional.absent();
+        try {
+            final EcsClientConfig clientConfig = createEcsClientConfig(clusterName, systemConfig, taskConfig); // ConfigException
+            try (final EcsClient client = ecsClientFactory.createClient(clientConfig)) { // ConfigException
+                final TaskDefinition td = findTaskDefinition(commandContext, client, taskConfig); // ConfigException
+                final Optional<ObjectNode> awsLogs = getAwsLogsConfiguration(td); // Nullable, ConfigException
+                // When RuntimeException is thrown by submitTask method, it will be handled and retried by BaseOperator, which is one of base
+                // classes of operator implementation.
+                final Task runTask = submitTask(commandContext, commandRequest, client, td); // ConfigException, RuntimeException
+                final ObjectNode currentStatus = createCurrentStatus(commandContext, commandRequest, clientConfig, runTask, awsLogs);
+                return createNextCommandStatus(commandContext, client, currentStatus);
+            }
+        }
+        catch (ConfigException e) {
+            logger.info("EcsCommandExecutor is not executed.", e);
+            return docker.run(commandContext, commandRequest); // fall back to DockerCommandExecutor
```

Comment:
I guess that fall back by ConfigException is a bit confused. 
And I have a question about fall back to DockerCommandExecutor is correct? Not SimpleCommandExecutor?

Reply:
Yes, fall back to Docker, not Simple. so, the comment on my change is correct.